### DISCUSSION TITLE
How can I disable or hide some resources by flag like 'admin.disable_flag'

### DISCUSSION THREAD

**Discussion by acentrist** [filamentphp/filament#3516] [Help] [answered]
How can I disable or hide some resources by flag like 'admin.disable_flag'?

**Comment by acentrist** [comment]
I just know how to get to do disable statically but I don't know how to do it dynamically.
```
protected static bool $shouldRegisterNavigation = false;
```

**Comment by Z3d0X** [comment]
Here are some helpful links
https://filamentphp.com/docs/2.x/admin/resources/getting-started#authorization
https://filamentphp.com/docs/2.x/admin/pages/getting-started#conditionally-hiding-pages-in-navigation
Self Promotion: https://github.com/z3d0x/filament-simple-permissions

**Comment by mikrosmile** [comment, marked as answer]
In resource page 
`    public static function canViewAny(): bool
    {
        return //any function which will check if user has access to;
    }`

This function will prevent from unauthorized user to access any records in related model (also can't globally search etc) and access by a direct link and removes from navigation. 

Function shouldRegisterNavigation just removes the page in navigation, but with direct link anyone can access

canView() is for a specific record to be accessible and viewed in table
canEdit() also can check if user has access to edit specific record
canCreate() so you understand the idea

**Reply by acentrist** [reply]
I got it, Thanks so match !
